src/components/PeopleFilters.tsx

```diff
@@ -1,93 +1,158 @@
-              className="button mr-1"
-              href="#/people?centuries=16"
-            >
-              16
-            </a>
+  const sexParam = searchParams.get('sex');
+  const allCenturies = searchParams.getAll('centuries');
+  const queryParam = searchParams.get('query') || '';
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=17"
-            >
-              17
-            </a>
+  const addCenturies = (century: string) => {
+    if (allCenturies.includes(century)) {
+      return [...allCenturies.filter(el => el !== century)];
+    }
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=18"
-            >
-              18
-            </a>
+    return [...allCenturies, century];
+  };
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=19"
-            >
-              19
-            </a>
+  return (
+    <>
+      <nav className="panel">
+        <p className="panel-heading">Filters</p>
 
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=20"
-            >
-              20
-            </a>
-          </div>
+        <p className="panel-tabs" data-cy="SexFilter">
+          <SearchLink
+            params={{ sex: null }}
+            className={cn({ 'is-active': sexParam === null })}
+          >
+            All
+          </SearchLink>
+          <SearchLink
+            params={{ sex: 'm' }}
+            className={cn({ 'is-active': sexParam === 'm' })}
+          >
+            Male
+          </SearchLink>
+          <SearchLink
+            params={{ sex: 'f' }}
+            className={cn({ 'is-active': sexParam === 'f' })}
+          >
+            Female
+          </SearchLink>
+        </p>
+        <div className="panel-block">
+          <p className="control has-icons-left">
+            <input
+              data-cy="NameFilter"
+              type="search"
+              className="input"
+              placeholder="Search"
+              value={queryParam}
+              onChange={handleChangeQuery}
+            />
 
-          <div className="level-right ml-4">
-            <a
-              data-cy="centuryALL"
-              className="button is-success is-outlined"
-              href="#/people"
-            >
-              All
-            </a>
-          </div>
+            <span className="icon is-left">
+              <i className="fas fa-search" aria-hidden="true" />
+            </span>
+          </p>
         </div>
-      </div>
+        <div className="panel-block">
+          <div
+            className="level is-flex-grow-1 is-mobile"
+            data-cy="CenturyFilter"
+          >
+            <div className="level-left">
+              <SearchLink
```

Comment:
the difference between all number centuries links is only number - you can map them instead of write each